src/App.tsx

```diff
@@ -1,26 +1,221 @@
+    };
+
+    setTodos(prev => [...prev, tempTodo]);
+
+    try {
+      const newTodo = await addTodos({
+        title: trimmedTitle,
+        completed: false,
+        userId: 3177,
+      });
+
+      setTodos(prev => prev.map(todo => (todo.id === 0 ? newTodo : todo)));
+      setSearchTerm('');
+    } catch {
+      setError('Unable to add a todo');
+      setTodos(prev => prev.filter(todo => todo.id !== 0));
+      throw new Error('Cant create new todos');
+    } finally {
+      setIsDisabledInput(false);
+
+      setTimeout(() => {
+        inputRef.current?.focus();
+      }, 0);
+    }
+  }
+
+  function removeTodos(todoId: number) {
+    return deleteTodo(todoId)
+      .then(() => {
+        setTodos(prev => prev.filter(todo => todo.id !== todoId));
+        inputRef.current?.focus();
+      })
+      .catch(() => {
+        setError('Unable to delete a todo');
+        throw new Error('Cant delete todos');
+      });
+  }
+
+  function changeTodo(todoId: number, title: string, completed: boolean) {
+    return patchTodos({ id: todoId, title, completed, userId: 3177 })
+      .then(() => {
+        setTodos(prev =>
+          prev.map(todo =>
+            todo.id === todoId ? { ...todo, title, completed } : todo,
+          ),
+        );
+      })
+      .catch(() => {
+        setError('Unable to update a todo');
+        throw new Error('Cant change todos');
+      });
+  }
+
+  function changeComplite() {
+    const isAllCompleted = todos.every(todo => todo.completed);
+    const updatedTodos = todos.map(todo => ({
+      ...todo,
+      completed: !isAllCompleted,
+    }));
+
+    setTodos(updatedTodos);
+
+    const todosToUpdate = todos.filter(
+      todo => todo.completed === isAllCompleted,
+    );
+
+    Promise.all(
+      todosToUpdate.map(todo =>
+        patchTodos({ ...todo, completed: !isAllCompleted }),
+      ),
+    )
+      .then(() => getTodos())
+      .then(setTodos)
+      .catch(() => {
+        setError('Unable to update todos');
+        throw new Error('Cant change all todos');
+      });
+  }
+
+  function filter(type: Filter) {
+    setFilterSelected(type);
+  }
+
+  function clearCompleted() {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    Promise.allSettled(completedTodos.map(todo => deleteTodo(todo.id)))
+      .then(results => {
+        const failedTodos = completedTodos.filter(
+          (todo, index) => results[index].status === 'rejected',
+        );
+
+        setTodos(prevTodos =>
+          prevTodos.filter(todo => {
+            if (!todo.completed) {
+              return true;
+            }
+
+            return failedTodos.some(failed => failed.id === todo.id);
+          }),
+        );
+
+        inputRef.current?.focus();
+
+        if (failedTodos.length > 0) {
+          setError('Unable to delete a todo');
+        }
+      })
+      .catch(() => {
+        setError('Unexpected error');
+      });
+  }
+
+  function clearError() {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const clearError = () => {
```